code_generation/quantity.py

```diff
@@ -129,3 +143,12 @@ def reserve_scope(self, scope):
             )
         )
         raise Exception
+
+    # if the shifted version of a quantity already exists in the input,
+    # this function can be used to register an
+    # branch from the input as a shifted version of a quantity
+    def register_external_shift(self, shift, external_name):
+        shift = "__" + shift
```

Comment:
better drop this line and add it in the config utility instead in order to be consistent with the implementation of the other variations